docs/source/daml/reference/generic-templates.rst

```diff
@@ -0,0 +1,60 @@
+.. Copyright (c) 2019 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+
+Generic Templates
+#################
+
+Sometimes different DAML templates have a common structure. Typically this occurs when there is some logic (usually in choices) that can be applied to many different underlying contracts, such as assets. Generic templates allow you to abstract template code from some of the data that would occur in the contracts. Let's see a simple example to understand what this means.
+
+Example: Generic Proposal
+*************************
+
+Suppose we want to model a propose and accept workflow. This means that a party can propose a contract to some number of receiving parties, and any one of them can accept it with the terms of that contract. We see this pattern occur frequently for different types of contracts. Of course, we could implement one proposal template for every underlying template. However this is tedious and error prone. Alternatively, we can write it once and for all using a generic template.
```

Comment:
"any one of them can accept" doesn't match your model. "the receiving parties can jointly accept" would be correct.

Reply:
Good point! Would something like the following do the trick?

Reply:
Ah true. I think to keep this example simple I'll just make it a proposal to a single party. Then I won't need any assertion because the `receiver` will need to exercise the `Accept` choice.